library/psa_crypto.c

```diff
@@ -1386,10 +1414,17 @@ static psa_status_t psa_start_key_creation(
      * the key creation fails or there is a power failure. */
     if( *p_drv != NULL )
     {
-        status = psa_find_se_slot_for_key( attributes, *p_drv,
-                                           &slot->data.se.slot_number );
-        if( status != PSA_SUCCESS )
-            return( status );
+        if( method != PSA_KEY_CREATION_REGISTER )
+        {
+            status = psa_find_se_slot_for_key( attributes, *p_drv,
+                                               &slot->data.se.slot_number );
+            if( status != PSA_SUCCESS )
+                return( status );
+        }
```

Comment:
Maybe a call to `p_check_slot` could be made here if the method is `PSA_KEY_CREATION_REGISTER`? Just to check that what the application is trying to register is sane.

Reply:
Yes, but not as is: `p_check_slot` might say “no, you can't create a key here because it's occupied”. I lean towards adding a `psa_key_creation_method_t` to `p_allocate` and `p_check_slot`, which would allow the driver to enforce constraints like “this slot is locked against import and can only contain internally-generated, non-exportable keys”.

Reply:
Done in the newest rebase. On a related note, I renamed `p_check_slot` to `p_validate_slot_number` in #202.